rocket_twin/systems/tank/tank_geom.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+from cosapp.base import System
+from OCC.Core.BRepGProp import brepgprop
+from OCC.Core.gp import gp_Pnt, gp_Vec, gp_XYZ
+from OCC.Core.GProp import GProp_GProps
+from pyoccad.create import CreateCylinder
+
+
+class TankGeom(System):
+    """Tank geometry.
+
+    Inputs
+    ------
+
+    Outputs
+    ------
+    rho [kg/m**3]: float,
+        mean density
+    shape: TopoDS_Solid
+        pyoccad model
+    """
+
+    def setup(self):
+
+        # Geometry
+        self.add_inward("radius", 1.0, desc="Base radius", unit="m")
+        self.add_inward("height", 1 / np.pi, desc="Height", unit="m")
+
+        # Weight
+        self.add_inward("weight", 1.0, desc="current weight", unit="kg")
+
+        # Position
+        self.add_inward("pos", 1 - 1 / np.pi, desc="base center z-coordinate", unit="m")
+
+        # Outputs
+        init = CreateCylinder.from_base_and_dir(gp_Pnt(0, 0, 0), gp_Vec(gp_XYZ(0, 0, 1)), 0)
```

Comment:
usefull ?

Reply:
Physically no, but the 'shape' inward needs a non empty solid as an initial value (otherwise we have an assignment error during copy)

Reply:
just use a basic shap to initiate it then

Reply:
Check